### DISCUSSION TITLE
Change DateTimeColumn to IntColumn

### DISCUSSION THREAD

**Discussion by ahuruFoundation** [simolus3/drift#1306] [Q&A]
Due this [issue](https://github.com/simolus3/moor/issues/215) (DateTimeColumn does not include milliseconds), I have to change it to IntColumn. But something concerns me that when of DateTimeColumn the value was 10 numbers , then when switch to IntColumn it becomes 13 numbers
![image](https://user-images.githubusercontent.com/79192128/124038780-d248e500-da55-11eb-9b30-0e0c4c290602.png)

How do I update the old values to add millseconds(append 000 to end?) to it during migration? I have checked complex migration but I don't know if ColumnTransformer can do it.

**Comment by simolus3** [comment]
A column transformer can do it! Something like this should work:

```dart
await m.alterTable(
  TableMigration(
    yourTable,
    columnTransformer: {
      yourTable.createDate: yourTable.createDate * Constant<int>(1000),
      yourTable.updatedAt: yourTable.updatedAt * Constant<int>(1000),
    },
  ),
);
```

But since you're not really changing the schema with a datetime to int conversion, you can also just use a regular update in your schema migration:

```dart
await update(yourTable).write(YourTableCompanion.custom(
  createDate: yourTable.createDate * Constant<int>(1000),
 updatedAt: yourTable.updatedAt * Constant<int>(1000),
));
```

**Reply by ahuruFoundation** [reply]
the columnTransformer works really well, but there is some problems with second approach.

For im using  a MillisDateConverter as you suggested in that [issue](https://github.com/simolus3/moor/issues/215#issuecomment-547832756), so that I can convert the number back to dateTime. 
`IntColumn get createDate => integer().map(const MillisDateConverter())();`
but it also lead to exception in the second approach:
`The argument type 'Expression<int?>' can't be assigned to the parameter type 'Expression<DateTime>?'.`

How can I address it?

**Reply by simolus3** [reply]
Yikes, I think that's a bug in the generator but I can't fix it without a breaking change release.

You can use `(yourTable.updatedAt * Constant<int>(1000)).dartCast()` to fix this for now.

**Reply by ahuruFoundation** [reply]
it works, thank you!
